### PR TITLE
missing prices: oracle-se2 and byol; some aurora

### PR DESCRIPTION
**Pull request by stanmluke**
**Description:**
Changes encompass the following:

- oracle-se2 (License Included) pricings were missing
- reordered enumeration of oracle instances, so permutations are more clear
- since oracle BYOL is the same for all types (as described by AWS), we now explicitly copy all byol
- minor change to sqlserver_web (includes multi-AZ), and, aurora (includes single-AZ).

Operationally, the result shows that oracle-se1 and oracle-se2 now yields the same number of prices;  and oracle-se/ee yields half that number (since they only have BYOL prices).

**Review:**

- [ ] Do the unit tests contain positive functionality tests as well as edge case negative tests?
- [ ] Can we monitor the performance/effectiveness of these changes?
- [ ] Did the code review consider failure modes and edge cases?
- [ ] Did the code review consider refactorings that would make this code more maintainable?
- [ ] Was this change approved by a module owner (if necessary)?

### CODE REVIEW

File: lib/amazon-pricing/rds-price-list.rb [hunk 1]
```diff
@@ -20,9 +20,12 @@ def initialize
...
       :postgresql => {:postgresql=>["standard","multiAZ"]},
       :oracle => {
-        :oracle_se1=>["li-standard","li-multiAZ","byol-standard","byol-multiAZ"],
+        # AWS: Oracle - Enterprise Edition, Standard Edition Two, Standard Edition, Standard Edition One.
+        # License Included Charges supported: Standard Edition One, Standard Edition Two
+        # BYOL Charges supported: do not vary by edition for BYOL Amazon RDS pricing.
         :oracle_se=>["byol-standard","byol-multiAZ"],
-        :oracle_ee=>["byol-standard","byol-multiAZ"]
+        :oracle_se1=>["li-standard","li-multiAZ"],
+        :oracle_se2=>["li-standard","li-multiAZ"]
```
Comment (ds2wang) [inline lib/amazon-pricing/rds-price-list.rb]: we are removing :oracle_ee=>["byol-standard","byol-multiAZ"]		 ?

Comment (stanmluke) [inline lib/amazon-pricing/rds-price-list.rb]: This is intentional:  see @@RESERVED_DB_WITH_SAME_PRICING2 as :oracle_se and :oracle_ee share the same pricings.

File: lib/amazon-pricing/definitions/database-type.rb [hunk 2]
```diff
@@ -105,7 +104,7 @@ def self.display_name(name)
   	def self.get_database_name
       [:mysql, :postgresql, :oracle_se1, :oracle_se, :oracle_ee, :sqlserver_ex, :sqlserver_web,
         :sqlserver_se, :sqlserver_ee, :aurora, :mariadb,
-        :oracle_se2 # oracle_se2 prices are copied, not collected
+        :oracle_se2 # oracle_se2 now collected (but not byol)
```
Comment (jeffzhou791) [inline lib/amazon-pricing/definitions/database-type.rb]: let's make this comment a little better (took me time to understand):

oracle_se2 license included prices are collected, and BYOL prices are copied

File: lib/amazon-pricing/definitions/rds-instance-type.rb [hunk 3]
```diff
@@ -85,8 +85,10 @@ def update_pricing(database_type, type_of_instance, json, is_multi_az, is_byol)
           end
         end
       end
-      # Copy prices from oracle_se to oracle_se2
+      # Copy byol prices from oracle_se to oracle_{ee,se1,se2}
       if database_type == :oracle_se && is_byol
+        update_pricing :oracle_ee,  type_of_instance, json, is_multi_az, is_byol
```
Comment (jeffzhou791) [inline lib/amazon-pricing/definitions/rds-instance-type.rb]: we are not collecting oracle_ee and oracle_se1?

**Comment by jeffzhou791** [comment]
you need to change version in Gemfile.lock as well

**Comment by jeffzhou791** [comment]
not entirely clear about your copying vs collecting criteria here. let's chat

File: lib/amazon-pricing/definitions/rds-instance-type.rb (hunk 3 above)
Comment (stanmluke) [inline lib/amazon-pricing/definitions/rds-instance-type.rb]: for oracle: we only collect for non-byol, since all byol prices are the same;  so we collect for oracle_se, oracle_se1, and, oracle_se2:
`    @@DB_Deploy_Types = {
      ...
      :oracle_se1   => [:standard, :multiaz, :byol, :byol_multiaz],
      :oracle_se2   => [:standard, :multiaz, :byol, :byol_multiaz],
      :oracle_se    => [:byol, :byol_multiaz],
      :oracle_ee    => [:byol, :byol_multiaz],
`
All of the above byol variants are the same, hence we copy (from oracle_se);  and that's why (for oracle) we don't collect from oracle_ee.  In summary, we collect {se, se1, se2}.

**Comment by jeffzhou791** [comment]
👍
